src/query/context.hpp

```diff
@@ -82,7 +91,9 @@ static_assert(std::is_move_assignable_v<ExecutionContext>, "ExecutionContext mus
 static_assert(std::is_move_constructible_v<ExecutionContext>, "ExecutionContext must be move constructible!");
 
 inline bool MustAbort(const ExecutionContext &context) noexcept {
-  return (context.is_shutting_down != nullptr && context.is_shutting_down->load(std::memory_order_acquire)) ||
+  return (context.transaction_status != nullptr &&
```

Comment:
This is probably more likely to cause perf problems -> one more reason to run mgbench 👀

Reply:
why if `is_shutting_down` doesn't create problems? Should be the same